toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -667,39 +693,38 @@ func calculateTotalPackages(packages []string, installRoot string) (totalPackage
 		// Search for the list of packages to be installed,
 		// it will be prefixed with a line "Installing:" and will
 		// end with an empty line.
-		inPackageList := false
 		for _, line := range splitStdout {
-			const (
-				packageListPrefix    = "Installing:"
-				packageNameDelimiter = " "
-			)
+			matches := listedPackageRegex.FindStringSubmatch(line)
+			if len(matches) != listMaxMatchLen {
+				// This line contains output other than a package information; skip it
+				continue
+			}
 
-			const (
-				packageNameIndex      = iota
-				extraInformationIndex = iota
-				totalPackageNameParts = iota
-			)
+			pkg := &repocloner.RepoPackage{
+				Name:         matches[installPackageName],
+				Version:      matches[installPackageVersion],
+				Architecture: matches[installPackageArch],
+				Distribution: matches[installPackageDist],
+			}
 
-			if !inPackageList {
-				inPackageList = strings.HasPrefix(line, packageListPrefix)
+			pkgID := pkg.ID()
+			if foundPackages[pkgID] {
+				logger.Log.Tracef("Skipping duplicate package: %s", line)
 				continue
-			} else if strings.TrimSpace(line) == "" {
-				break
 			}
+			foundPackages[pkgID] = true
 
-			// Each package to be installed will list its name, followed by a space and then various extra information
-			pkgSplit := strings.SplitN(line, packageNameDelimiter, totalPackageNameParts)
-			if len(pkgSplit) != totalPackageNameParts {
-				err = fmt.Errorf("unexpected TDNF package name output: %s", line)
-				return
-			}
+			logger.Log.Debugf("Added installedPackages entry for: %v-%v.%v.%v", pkg.Name, pkg.Version, pkg.Distribution, pkg.Architecture)
 
-			allPackageNames[pkgSplit[packageNameIndex]] = true
+			installedPackages.Repo = append(installedPackages.Repo, pkg)
 		}
 	}
 
-	totalPackages = len(allPackageNames)
-	logger.Log.Debugf("All packages to be installed (%d): %v", totalPackages, allPackageNames)
+	logger.Log.Debugf("Total number of packages to be installed: %d", len(installedPackages.Repo))
+
+	// Write out JSON file with list of packages included in the image
+	err = jsonutils.WriteJSONFile("/installroot/image_pkg_manifest_temp1.json", installedPackages)
```

Comment:
Let's put that path to a public named constant, so that it can be used in `ExtractFile` from a different module instead of using a raw string in multiple places.

Reply:
Sure, done